livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py

```diff
@@ -54,6 +72,56 @@ async def chat(
 
         return LLMStream(cmp, fnc_ctx)
 
+    def _sync_image_cache(self, history: llm.ChatContext):
+        video_frame_images = [
+            img
+            for msg in history.messages
+            for img in msg.images
+            if isinstance(img, llm.ChatMessageVideoFrameImage)
+        ]
+        seen_images = set(video_frame_images)
+        (w, h) = ImageDetailsResizeValues[self._opts.image_detail]
+        encode_options = images.EncodeOptions(
+            format="JPEG",
+            resize_options=images.ResizeOptions(
+                width=w, height=h, strategy="center_aspect_fit"
+            ),
+        )
+        for img in video_frame_images:
```

Comment:
wouldn't this reencode everything each time?

it seems like the algo would be better to:
* in ChatContext, allow users to specify the following
  * max_images
  * max_messages
* truncate context before syncing image cache
* while syncing, iterate through the entire list
* for any image not already cached, encode it
* while iterating, count how many images there has been
* truncate earlier images

Reply:
Fixed the logic bug here. In this PR I purposely don't do any extra management of the context and leave that up to users. I'm not sure if users might want to more sophisticated treatment of images. Idk, maybe there is a use case where they always want the first image and then the last N images.